Assets/Scripts/Services/History/ModeAction.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Controllers;
+using Services.History;
+using System;
+
+public class ModeAction : UserAction {
+	public static Action<ConnectionMode> changeModeAction;
```

Comment:
`ModeAction` is basically the same as `NodeTraverseModeAction`. They should be generified into `ModeChangeAction` used for any two value enum switch action.